### DISCUSSION TITLE
Access family provider without passing values once initialized

### DISCUSSION THREAD

**Discussion by haaris94** [rrousselGit/riverpod#3697] [Q&A]
How can I access a family provider without passing values again after it has been initialized?

For example, on the `HomeScreen`, I initialize the provider like this:

```dart
.read(outlineControllerProvider(
  articleRequirements: articleRequirements,
  aiOptions: aiOptions,
).notifier).loadOutline();
```

After initialization, I want to access the provider in any of the child widgets without having to pass the values again. Currently, I am passing the provider's properties through the `constructor` of all child widgets that need a reference to `outlineControllerProvider`.

**Comment by snapsl** [comment]
Simple answer: You can't. 
How is the family provider supposed to know which instance you want to access if there are several family provider instances?
You need to revaluate if you need a family provider.
Another option is to use a [scoped provider](https://riverpod.dev/docs/concepts/scopes#subtree-scoping) which lets you access a specific family provider instance.

**Reply by haaris94** [reply]
Can you give me a simple code example of overriding the above a family AsyncNotifier provider? I cannot find any code examples.

**Comment by gdurandrexel** [comment]
Since you can't do that, you may acheive what you want by having a build method that returns a dummy value and have a method  initialize(articleRequirements, aiOptions) that updates the provider state and that you call once in your main screen. Your provider must be kept alive somehow (either watched or listened to in the home screen or declared with keepAlive=true).
